content/learn/book/modular-architecture/project-organization.md

```diff
@@ -0,0 +1,107 @@
++++
+title = "Project Organization"
+insert_anchor_links = "right"
+[extra]
+weight = 1
++++
+
+As your Bevy project grows, storing it all in a single `main.rs` file will become frustrating!
+But how should you split it apart, and what are your tools for doing so?
+
+Thankfully, the answers to these questions are pretty simple:
+
+- split your code based on the "domain" it covers
+  - think about where you might want to reuse, or selectively disable bits of functionality
+  - code that changes together should live together
+  - splits like "world generation" and "combat" and "main menu" are great
+  - don't organize your modules by e.g. "components" and "systems"!
+- don't try to get it right from the very start
+  - split things up gradually, as areas become complex and unwieldy
+  - you'll have a much better idea of the shape of your program then
+  - and it stops you from wasting time at the start of your project
+- take advantage of Rust's modules, crates and visibility to organize your code
+  - be mindful of [orphan rules] when architecting multi-crate projects
+- one [plugin] per module, and one plugin per crate (which may nest multiple plugins) is a good default
+  - placing these at the top of your files can act as a simple, consistent "table of contents"
+
+[orphan rules]: https://doc.rust-lang.org/reference/items/implementations.html#orphan-rules
+[plugin]: ./plugins.md
+
+## Modules
+
+Rust's [module system] is flexible and easy to use,
+allowing you to split code across files.
+
+Generally, we find that splitting files that are beyond about 1000 lines is
+about the right size for Rust and Bevy projects.
+Too small and you'll end up constantly swapping between files,
+and too large and finding code within a file becomes frustrating.
+
+Modules within files aren't particularly useful,
+but they can be used to control visibility, or enable code folding.
+
+[module system]: https://doc.rust-lang.org/book/ch07-02-defining-modules-to-control-scope-and-privacy.html
+
+## Crates
+
+As your project grows even further,
+you should consider splitting it into multiple [crates] using [workspaces].
+
+Using crates is helpful when reusing code between projects,
```

Comment:
```suggestion
Using crates is helpful when reusing code between projects, formalizing APIs for specific domains,
```

The Crates section talks a decent amount about performance, but I think its worth connecting it back to the continued stair-step from main -> main/lib -> modules & plugins -> crates -> crates you use across your own projects -> publishing crates on crates.io, which can be thought of as a growing formalization of a given domain's API. "this is my splash screen" -> "this is the splash screen I use on all my projects" -> "this is a splash screen I made for others to use in their projects"